### PR TITLE
[Feature] support `LZO` compress on File Read

### PR DESCRIPTION
**Pull request by liunaijie**
## Purpose of this pull request
support compress on local text file read. 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Good pr, let's waiting CI/CD

**Review by TaoZex — COMMENTED** [review]
LGTM
But the doc needs to be modified.

**Comment by liunaijie** [comment]
> LGTM But the doc needs to be modified.

doc updated.

**Comment by liunaijie** [comment]
updated json reader.
updated document

**Comment by liunaijie** [comment]
@TaoZex @TyrantLucifer  PTAL

**Comment by liunaijie** [comment]
pls re-trigger the CI as i update more formart reader

**Comment by TyrantLucifer** [comment]
> pls re-trigger the CI as i update more formart reader

Done, let's waiting CI/CD.

**Comment by liunaijie** [comment]
pls trigger the CI

**Comment by liunaijie** [comment]
@TyrantLucifer @TaoZex  PTAL

**Comment by ic4y** [comment]
The documentation for connectors depending on the file-base package, such as hdfsFile, oss, and s3, likely needs updating.

**Comment by liunaijie** [comment]
> The documentation for connectors depending on the file-base package, such as hdfsFile, oss, and s3, likely needs updating.

@ic4y i will create another pr (add test case and update doc) for connectors depending on the file-base

**Comment by ic4y** [comment]
> > The documentation for connectors depending on the file-base package, such as hdfsFile, oss, and s3, likely needs updating.
> 
> @ic4y i will create another pr (add test case and update doc) for connectors depending on the file-base

It's best to complete this within this PR, as it's part of the whole.

**Comment by liunaijie** [comment]
@ic4y updated related connector and document, PTAL, thanks

**Comment by ic4y** [comment]
Waiting for CI/CD.

**Comment by liunaijie** [comment]
failed on `ClusterFaultToleranceTwoPipelineIT.testTwoPipelineStreamJobRestoreIn2NodeMasterDown:732 » ConditionTimeout`

can you help re-trigger the CI, thanks

**Comment by liunaijie** [comment]
i can see the CI error has been fixed in another pr, can you help trigger this CI again, thanks.

**Comment by liunaijie** [comment]
wait for this issue fix
https://github.com/apache/seatunnel/issues/5342

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_json_lzo_to_console.conf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          c_timestamp = timestamp
+        }
+      }
+    }
+
+  }
+}
+
+sink {
+  Console {
```
Comment (TyrantLucifer) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_json_lzo_to_console.conf:86]: Please add assert connector to verify the data is valid.

**Review by TyrantLucifer — DISMISSED** [review]

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_json_lzo_to_console.conf (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_json_lzo_to_console.conf:86]: > Please add assert connector to verify the data is valid.

@liunaijie  You can reference [`seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/func_system.conf`](https://github.com/apache/seatunnel/blob/d308e2773321ca7c6b0c27be51ad122a06b23ea9/seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/resources/sql_transform/func_system.conf#L4)

Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_json_lzo_to_console.conf:86]: Why you not use `equals_to` to check the value of the row?

image

Comment (liunaijie) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-local-e2e/src/test/resources/json/local_file_json_lzo_to_console.conf:86]: updated.
previous the json file was generated by fake source with random values, so i am not apply the equal check.
Now i update the json file, and add the equal check

**Review by EricJoy2048 — COMMENTED** [review]
I am sorry, In the Apache project, it is not allowed to publish binary files in the source code package (because the content of the binary file is not visible and cannot be modified. If there is a problem in the file, no one can fix it), so an alternative solution is to generate the binary file through the program (the generated code is visible, and if there is a problem, the code can also be modified to fix the problem).

**Comment by Hisoka-X** [comment]
@liunaijie I updated test case for this. Please take a look when you have free time.

**Comment by Hisoka-X** [comment]
The test case have some bug will be fixed in https://github.com/apache/seatunnel/pull/5622

**Comment by liunaijie** [comment]
> @liunaijie I updated test case for this. Please take a look when you have free time.

@Hisoka-X  cool, thanks. the test case change looks good to me, but i find you change the `discoverTestContainers` method, look this change is only for testing, should't push.

**Comment by Hisoka-X** [comment]
> > @liunaijie I updated test case for this. Please take a look when you have free time.
> 
> @Hisoka-X cool, thanks. the test case change looks good to me, but i find you change the `discoverTestContainers` method, look this change is only for testing, should't push.

oh, good catch. Reverted.

**Review by EricJoy2048 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]
Thanks @liunaijie . LGTM

**Pull request merged**